Resources/public/js/views/services/ez-trashviewservice.js

```diff
@@ -223,6 +223,55 @@ YUI.add('ez-trashviewservice', function (Y) {
         },
 
         /**
+         * Restores trash items
+         *
+         * @method _restoreItems
+         * @protected
+         * @param {Object} e  restoreItems event facade
+         * @param {Array} e.trashItemsId  List of trashItems id to be restored
+         *
+         * @return {Boolean}
+         */
+        _restoreItems: function (e) {
+            var contentService = this.get('capi').getContentService(),
+                tasks = new Y.Parallel(),
+                service = this,
+                notificationNamePrefix = "restoreTrashItem-";
+
+            Y.each(e.trashItemsId, tasks.add(function (trashItemId) {
+                contentService.recover(trashItemId, function (error) {
```

Comment:
it would be cleaner if the trash item model have a `restore` method. That also means the `restoreItems` should provide the trash items models, not only the ids.